tests/undefined_symbol_in_sdfg_test.py

```diff
@@ -11,51 +11,111 @@
+    
+    # Add arrays to SDFG
+    sdfg.add_array('A', [20, 20], dace.float64)
+    sdfg.add_array('B', [20], dace.float64)
+    sdfg.add_transient('tmp', [20, undefined_dim], dace.float64)
+    
+    # Create state and add nodes
+    state = sdfg.add_state()
     
     # Validate SDFG - it should pass validation at this point
     sdfg.validate()
     
-    # Try to generate code - this should fail
+    # Try to generate code - this should fail because of the undefined symbol
     with pytest.raises(TypeError, match="undefined symbol"):
         sdfg.compile()
 
 
-def test_undefined_symbol_propagation():
-    """Tests that UndefinedSymbol propagates through symbolic expressions."""
+def test_undefined_symbol_in_unused_dimension():
+    """Tests that an UndefinedSymbol in an unused dimension allows code generation to proceed."""
+    
+    # Create an SDFG manually
+    sdfg = dace.SDFG('undefined_symbol_unused_test')
+    
+    # Define symbols
+    undefined_dim = symbolic.UndefinedSymbol()
+    
+    # Add arrays to SDFG
+    sdfg.add_array('A', [20, 20], dace.float64)
+    sdfg.add_array('B', [20], dace.float64)
+    
+    # Add transient with undefined dimension in the first position
+    sdfg.add_transient('tmp', [undefined_dim, 20], dace.float64)
+    
+    # Create state
+    state = sdfg.add_state()
+    
+    # Create access nodes
+    A = state.add_read('A')
+    tmp = state.add_access('tmp')
+    B = state.add_write('B')
+    
+    # Add MapEntry and MapExit nodes
+    map_entry, map_exit = state.add_map('compute', dict(i='0:20'))
+    
+    # Add tasklets
+    tasklet_write = state.add_tasklet('write', {'a'}, {'b'}, 
+                                     'b = a')
+    tasklet_read = state.add_tasklet('read', {'a'}, {'b'}, 
+                                    'b = a')
+    
+    # Connect everything with only using the defined dimension (avoid the undefined one)
+    state.add_edge(A, None, tasklet_write, 'a', dace.Memlet.simple('A[0, 0]'))
+    state.add_edge(tasklet_write, 'b', tmp, None, dace.Memlet.simple('tmp[0, 0]'))
+    state.add_edge(tmp, None, tasklet_read, 'a', dace.Memlet.simple('tmp[0, 0]'))
+    state.add_edge(tasklet_read, 'b', B, None, dace.Memlet.simple('B[0]'))
+    
+    # Validate SDFG - it should pass validation
+    sdfg.validate()
+    
+    # This should compile successfully since the undefined dimension is never used
+    sdfg.compile()
+
+
+def test_undefined_symbol_value_assignment():
+    """Tests that an UndefinedSymbol can be assigned a value before code generation."""
+    
+    # Create an SDFG manually
+    sdfg = dace.SDFG('undefined_symbol_assignment_test')
+    
+    # Define symbols
+    undefined_dim = symbolic.UndefinedSymbol()
     
-    # Create expressions with undefined symbols
-    a = symbolic.symbol('a')
-    b = symbolic.symbol('b')
-    undefined = symbolic.UndefinedSymbol()
+    # Add arrays to SDFG
+    sdfg.add_array('A', [20, 20], dace.float64)
```

Comment:
The SDFG created here is not very interesting. It just defines some data containers but does nothing with them.